desktop/core/src/desktop/js/api/utils.ts

```diff
@@ -187,8 +192,10 @@ export const post = <T, U = unknown>(
     const { cancelToken, cancel } = getCancelToken();
     let completed = false;
 
+    const encodeData = options?.qsEncodeData == undefined || options?.qsEncodeData;
```

Comment:
undefined check, to enabled qsEncodeData by default as expected by the existing calls to post

Reply:
The alternative would be to change the type of "data" to `URLSearchParams` and adjust all the other code that calls this function.

Reply:
... or, if you're indeed using URLSearchParams.. `const encode = !(data instanceof URLSearchParams)`

Reply:
Reasoning is that this is specific to how Axios works (https://axios-http.com/docs/urlencoded)

Reply:
All the Hue APIs needs qs encoding, QP APIs are expecting the default JSON object. So what should be the approach?

Reply:
I am a bit confused. Is the '/jobbrowser/query-store/api/impala/queries' not a Hue API?

Reply:
Nop, /jobbrowser/query-store proxies the request to query store (Alias QP).
https://github.com/cloudera/hue/blob/master/apps/jobbrowser/src/jobbrowser/urls.py#L82

Reply:
I see. So next question, are you saying that the default `post` calls in Hue are encoding the post data? That doesn't seem right.

Reply:
What matters is that all Hue APIs behave consistently (whether they proxy or not). My suggestion here is to adjust the encoding in the Hue backend (`jobbrowser_api2.query_store_api`) to convert it to whatever the QP service expects.

Reply:
To answer @bjornalm's question... I believe this comes from Hue historically wanting application/x-www-form-urlencoded requests, the way to handle that with axios post is described [here](https://axios-http.com/docs/urlencoded). But yeah, perhaps that can be changed as well...

So, I guess what it comes down to is that we need to decide on an approach and stick with it throughout.

Reply:
@sreenaths as far as the Hue UI is concerned it is a Hue API endpoint. The UI shouldn't know about the implementation details (proxy).

Reply:
I agree that the APIs should be consistent. But I think we are from from that today. The reason I was so confused about this default encoding is that it is not used by the file browser. So the file browsing APIs are expecting unencoded post data.. which I think makes sense. So what we probably should do is to change the APIs that do expect form data to be encoded, unless there is a good reason for them to do that in which case we have to support both scenarios.

Reply:
@sreenaths That is obviously not a task for this PR :-)

Reply:
I see now that the Hue API endpoint for the QP is using `json.loads(request.body)` [here](https://github.com/cloudera/hue/blob/master/apps/jobbrowser/src/jobbrowser/api2.py#L187). Searching through the Hue codebase it seems to be the only API endpoint that expects json directly from request.body and it feels reasonable to support this.

How about renaming `options.qsEncodeData` to `options.jsonBody` for now? Later we should look into making jsonBody the default and adjust where it breaks.

Reply:
We can do that, although naming it `options.sendJsonBody` can be a bit confusing if the data is just a simple (non-json) string that we don't want to url encode. Since this function will continue to encode the post data by default using qs.stringify I think the originally suggested `options.qsEncodeData` is also good for now:

`const {qsEncodeData = true } = options;`

But this was enlightening! I agree that when we (or a future AI) later harmonize the HUE APIs the default should be not to encode the post data.